core/account-id/src/errors.rs

```diff
@@ -22,22 +24,32 @@ impl fmt::Display for ParseAccountError {
 }
 
 /// A list of errors that occur when parsing an invalid Account ID.
-#[derive(Eq, Hash, Clone, Debug, PartialEq)]
+#[non_exhaustive]
+#[derive(Eq, Clone, Debug, PartialEq)]
 pub enum ParseErrorKind {
     TooLong,
     TooShort,
     Invalid,
```

Comment:
probably makes sense to add docs to variants here

Reply:
Feels like we might want to split `Invalid` into more fine graind componets: InvalidChar, DuplicateSeparator, TrailingSeparator. doing that aftre release would technically be a breaking change.

Reply:
Already did this on a local branch 😄 , I went with the following names though - `HasConsecutiveSeparators`, `HasUnterminatedSeparators` and `HasInvalidChars`.

Reply:
sgtm! 

Another option for `ConsecutiveSeparators` and `TrailingSeparator` is to go dual. The rule isn't "you can't have two separators, and a trailing separator, and a leading separator". The rule is rather "separators separate non-empty parts". So the actual error here is `EmptyPart`. 


:thinking: having looked at the impl again, I do think we shouldn't try to distinguish separator-related errors, as `..` is erroneous for three reasons. That is, I am pretty sure there should be only one error variant here. So, either `EmptyPart` if we focus on parts, or `InvalidSeparator` if we focus on separators.

Reply:
The error description for `InvalidSeparator` I feel would be lacking to some degree. `EmptyPart` sounds good to me.

Reply:
So, `EmptyPart` is good for all error cases in `..` but not so much for `__` because according to the spec:

> **Account ID part** consists of lowercase alphanumeric symbols separated by either `_` or `-`. [ <sub> (doc) </sub> ](https://docs.near.org/docs/concepts/account#account-id-rules)

So, `_` and `-`, don't delimit a part. They're just a sort-of “spacer” within a part. Meaning, while `..` has 3 empty parts, `__` has one _non-empty, though invalid_ part.

Back to naming: maybe `ExtraneousSeparator`, `ExcessiveSeparator` or `RedundantSeparator`? 😪

I'm voting `RedundantSeparator`.

Reply:
Implemented in https://github.com/near/nearcore/pull/5660